php/class-wp-seo-social-settings.php

```diff
@@ -0,0 +1,708 @@
+							'field' => 'single_' . $post_type->name . '_og_description',
+							'type'  => 'textarea',
+						),
+					),
+					array(
+						'id'       => 'single_' . $post_type->name . '_og_image',
+						'title'    => __( 'Open Graph Image', 'wp-seo-social' ),
+						'section'  => 'single_' . $post_type->name,
+						'args'     => array(
+							'field' => 'single_' . $post_type->name . '_og_image',
+							'type'  => 'image',
+						),
+					),
+					array(
+						'id'       => 'single_' . $post_type->name . '_og_type',
+						'title'    => __( 'Open Graph Type', 'wp-seo-social' ),
+						'section'  => 'single_' . $post_type->name,
+						'args'     => array(
+							'field' => 'single_' . $post_type->name . '_og_type',
+							'type'  => 'dropdown',
+							'boxes' => array(
+								'website' => 'Website',
+								'article' => 'Article',
+							),
+						),
+					),
+				)
+			);
+		}
+		$archived_post_settings = array();
+		foreach ( $this->archived_post_types as $post_type ) {
+			$archived_post_settings = array_merge(
+				$archived_post_settings,
+				array(
+					array(
+						'id'       => 'archive_' . $post_type->name . '_og_title',
+						'title'    => __( 'Open Graph Title Tag Format', 'wp-seo-social' ),
+						'section'  => 'archive_' . $post_type->name,
+						'args'     => array(
+							'field' => 'archive_' . $post_type->name . '_og_title',
+						),
+					),
+					array(
+						'id'       => 'archive_' . $post_type->name . '_og_description',
+						'title'    => __( 'Open Graph Description Tag Format', 'wp-seo-social' ),
+						'section'  => 'archive_' . $post_type->name,
+						'args'     => array(
+							'field' => 'archive_' . $post_type->name . '_og_description',
+							'type'  => 'textarea',
+						),
+					),
+					array(
+						'id'       => 'archive_' . $post_type->name . '_og_image',
+						'title'    => __( 'Open Graph Image', 'wp-seo-social' ),
+						'section'  => 'archive_' . $post_type->name . '_og_description',
+						'args'     => array(
+							'field' => 'archive_' . $post_type->name . '_og_image',
+							'type'  => 'image',
+						),
+					),
+					array(
+						'id'       => 'archive_' . $post_type->name . '_og_type',
+						'title'    => __( 'Open Graph Type', 'wp-seo-social' ),
+						'section'  => 'archive_' . $post_type->name,
+						'args'     => array(
+							'field' => 'archive_' . $post_type->name . '_og_type',
+							'type'  => 'dropdown',
+							'boxes' => array(
+								'website' => 'Website',
+								'article' => 'Article',
+							),
+						),
+					),
+				)
+			);
+		}
+		$all_settings = array_merge(
+			$social_settings,
+			$taxonomy_settings,
+			$post_settings,
+			$archived_post_settings
+		);
+		foreach ( $all_settings as $setting ) {
+			add_settings_field(
+				$setting['id'],
+				$setting['title'],
+				array( 'WP_SEO_Settings', 'field' ),
+				'wp-seo',
+				$setting['section'],
+				$setting['args']
+			);
+			if ( ! isset( $setting['args']['type'] )
```

Comment:
It might be worth adding a description above this if statement to describe what you are trying to do.  To me it seems like every field will be included here haha.